From 5f48f940ee429a096db98c2e23c841f9811119c5 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 16 Jul 2021 06:54:13 +0300
Subject: [PATCH] [SW-0] goya2: remove redundant code

'goya2_arc' is redundant. Instead 'greco_arc' is used.
So removed 'goya2_arc'.

Change-Id: I235c89fe635dd71ef4701bbe2eeae3364e269841
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/goya2_arc/common.c b/goya2_arc/common.c
deleted file mode 100644
index b6ea28d..0000000
--- a/goya2_arc/common.c
+++ /dev/null
@@ -1,169 +0,0 @@
-// SPDX-License-Identifier:     GPL-2.0+
-/*
- * Copyright (C) 2020 HabanaLabs Ltd.
- * All Rights Reserved.
- *
- */
-
-#include <unistd.h>
-#include <efuse.h>
-#include <pcie.h>
-#include <psoc_efuse_regs.h>
-#include <psoc_global_conf_masks.h>
-#include <arcboot.h>
-
-#define IATU_BASE				(mmPCIE_DBI_BASE & 0xFFFFFF)
-#define IATU_REGION_CTRL_1_OFF_INBOUND		(IATU_BASE + 0x100)
-#define IATU_REGION_CTRL_2_OFF_INBOUND		(IATU_BASE + 0x104)
-#define IATU_LWR_TARGET_ADDR_OFF_INBOUND	(IATU_BASE + 0x114)
-#define IATU_UPPER_TARGET_ADDR_OFF_INBOUND	(IATU_BASE + 0x118)
-
-uint32_t get_efuse_read_reg(void)
-{
-	return mmPSOC_EFUSE_READ;
-}
-
-uint32_t get_mem_cfg_mask(void)
-{
-	return PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_FLD_SIZE_MASK;
-}
-
-uint32_t get_fld_size_shift(void)
-{
-	return PSOC_GLOBAL_CONF_SPI_DMA_CTRL_FLD_SIZE_SHIFT;
-}
-
-void update_warm_reset_config(void)
-{
-	uint32_t reg;
-	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
-	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
-	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_PSOC_MASK;
-	regmap_writel(reg, mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
-}
-
-uint8_t get_sris_mode_from_strap(void)
-{
-	uint32_t boot_strap;
-
-	boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
-
-	return !!(boot_strap & PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_SRIS_MODE_MASK);
-}
-
-void enable_flr_handling_fsm(void)
-{
-	uint32_t flr_ctrl = regmap_readl(mmPCIE_AUX_FLR_CTRL);
-
-	flr_ctrl |= (1 << PCIE_AUX_FLR_CTRL_FSM_EN_SHIFT);
-	regmap_writel(flr_ctrl, mmPCIE_AUX_FLR_CTRL);
-}
-
-void pcie_config_iatu(void)
-{
-	regmap_writel(0x300000, mmPCIE_AUX_DBI);
-	regmap_writel(0xF0000000, IATU_LWR_TARGET_ADDR_OFF_INBOUND);
-	regmap_writel(0x7F, IATU_UPPER_TARGET_ADDR_OFF_INBOUND);
-	regmap_writel(0, IATU_REGION_CTRL_1_OFF_INBOUND);
-	regmap_writel(0xC0080000, IATU_REGION_CTRL_2_OFF_INBOUND);
-}
-
-void pcie_phy_para_sel_set(void)
-{
-	regmap_writel(1, mmPCIE_PHY_CR_PARA_SEL_0);
-	regmap_writel(1, mmPCIE_PHY_CR_PARA_SEL_1);
-}
-
-void pcie_wait_for_sram_init_done(void)
-{
-	uint8_t sram_init_done = 0;
-
-	do {
-		sram_init_done =
-			regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_0);
-	} while (sram_init_done != 1);
-
-	sram_init_done = 0;
-	do {
-		sram_init_done =
-			regmap_readl(mmPCIE_PHY_PHYX_SRAM_INIT_DONE_1);
-	} while (sram_init_done != 1);
-}
-
-void pcie_phy_set_sram_ext_ld_done(void)
-{
-	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_0);
-	regmap_writel(1, mmPCIE_PHY_PHYX_SRAM_EXT_LD_DONE_1);
-}
-
-void pcie_phy_set_mpll_a_b_ssc_enable(void)
-{
-
-	pcie_update_reg_range(mmPCIE_PHY_PHYX_MPLLB_SSC_EN_0,
-			mmPCIE_PHY_PHYX_MPLLB_SSC_EN_1, 0x1);
-	pcie_update_reg_range(mmPCIE_PHY_PHYX_MPLLA_SSC_EN_0,
-			mmPCIE_PHY_PHYX_MPLLA_SSC_EN_1, 0x1);
-}
-
-void pcie_phy_set_mpll_a_b_mul_bw(uint8_t separate_reference)
-{
-	uint32_t reg;
-
-	if (separate_reference) {
-		reg = 0xA0;
-		regmap_writel(0x00250025,
-			      mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH);
-		regmap_writel(0x00250025,
-			      mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH);
-	} else {
-		reg = 0x50;
-		regmap_writel(0x01C301C3,
-			      mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH);
-	}
-
-	regmap_writel(reg, mmPCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER);
-	regmap_writel(0x7D, mmPCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER);
-}
-
-void pcie_phy_set_ext_tx_rx_ctrl(void)
-{
-	pcie_update_reg_range(mmPCIE_PHY_EXT_TX_DCC_CTRL_RANGE_G1,
-			mmPCIE_PHY_EXT_TX_DCC_CTRL_RANGE_G4, 0xC);
-
-	pcie_update_reg_range(
-		mmPCIE_PHY_PROTOCOL0_EXT_RX_CDR_VCO_CONFIG_G1_0,
-		mmPCIE_PHY_PROTOCOL0_EXT_RX_CDR_VCO_CONFIG_G1_7, 0x1);
-	pcie_update_reg_range(
-		mmPCIE_PHY_PROTOCOL0_EXT_RX_CDR_VCO_CONFIG_G2_0,
-		mmPCIE_PHY_PROTOCOL0_EXT_RX_CDR_VCO_CONFIG_G2_7, 0x1);
-}
-
-void pcie_phy_set_protocol0_ext_tx_rx(void)
-{
-	regmap_writel(0x30,
-		      mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G1_SLICE0);
-	regmap_writel(0x30,
-		      mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G2_SLICE0);
-	regmap_writel(0x30,
-		      mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G3_SLICE0);
-	regmap_writel(0x30,
-		      mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G4_SLICE0);
-	pcie_update_reg_range(
-		mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_POST_G1,
-		mmPCIE_PHY_PROTOCOL0_EXT_TX_EQ_POST_G4, 0);
-
-	regmap_writel(PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF,
-			mmPCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3);
-	regmap_writel(PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF,
-			mmPCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4);
-}
-uint8_t get_srns_mode_from_strap(uint32_t boot_strap)
-{
-	return !!(boot_strap & PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_EN_MASK);
-}
-
-uint8_t get_ssc_en_from_strap(uint32_t boot_strap)
-{
-	return (!!(boot_strap &
-			PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L_BTL_ROM_EN_MASK));
-}
diff --git a/goya2_arc/soc.h b/goya2_arc/soc.h
deleted file mode 100644
index 8ef900d..0000000
--- a/goya2_arc/soc.h
+++ /dev/null
@@ -1,41 +0,0 @@
-/*
- * Copyright (C) 2017 HabanaLabs Ltd.
- *   Author: Noam Camus <noamikong@gmail.com>
- *
- * SPDX-License-Identifier:     GPL-2.0+
- */
-
-#ifndef _SOC_H_
-#define _SOC_H_
-#include <goya2_regs.h>
-
-#define CPU_FREQ	500000000
-#define CPU_FREQ_US	500	/* CPU_FREQ / 1000000 */
-#define PREBOOT_ADDR_H	0x7F
-#define PREBOOT_ADDR_L	0xF0000000
-
-#define FW_SIZE					65535
-#define SIG_SIZE				384
-#define KEY_SIZE				392
-#define BLOB_START				4
-
-#define SPI_FLASH_BLOCK_SIZE			1024
-
-/* 16MB for Greco */
-#define FLASH_VALID_MASK			0xFFFFFF
-#define FLASH_VALID_OFFSET			0
-#define FLASH_VALID_MAX_SIZE			0x1000000
-
-#define HL_SRAM_BASE				0x7FF0000000ULL
-#define HL_SP_SRAM_BASE            		0x7FFBFE0000ULL
-#define HL_SDRAM_BASE				0x10000000
-#define HL_FLASH_OFFSET				0xA000000
-#define HL_ICCM_OFFSET				0xB020000
-#define HL_SBROM_OFFSET				0xB040000
-#define HL_FWSRAM_OFFSET			0xB050000
-#define HL_SCRATCHPAD_SRAM_OFFSET		0xBFE0000
-#define HL_PCIE_FWSRAM_OFFSET			0xBFF0000
-
-#define HL_CFG_BASE				0xFC000000
-
-#endif /* _SOC_H_ */
